### PR TITLE
Update list subnet

### PR DESCRIPTION
**Pull request by cryptoAtwill**
Fix list subnets parsing error.

### CODE REVIEW

File: src/lotus/message/ipc.rs [hunk 1]
```diff
@@ -36,21 +39,28 @@ pub struct IPCReadSubnetActorStateResponse {
...
     /// Collateral staked in the subnet.
+    #[serde(rename(deserialize = "Stake"))]
     #[serde(deserialize_with = "deserialize_token_amount_from_str")]
+    #[serde(serialize_with = "serialize_token_amount_to_atto")]
     pub stake: TokenAmount,
     /// Circulating supply available in the subnet.
+    #[serde(rename(deserialize = "CircSupply"))]
     #[serde(deserialize_with = "deserialize_token_amount_from_str")]
-    #[serde(rename = "CircSupply")]
+    #[serde(serialize_with = "serialize_token_amount_to_atto")]
```
Comment (adlrocha) [inline src/lotus/message/ipc.rs:60]: Why do we need this additional deserialization step? Can't we deserialize directly into token amount (and change SubnetInfo if needed)

File: src/lotus/message/tests.rs [hunk 2]
```diff
@@ -78,3 +83,41 @@ fn test_token_amount_from_str() {
...
+#[test]
+fn test_subnet_info_from_str() {
+    set_current_network(Network::Testnet);
+
+    let raw_str = r#"
+    {
+        "ID": {
+            "Parent": "/root",
+            "Actor": "t01002"
+        },
```
Comment (adlrocha) [inline src/lotus/message/tests.rs:109]: We should check a grater value to see if it works with overflows (I had to fix a similar bug in deserialization  a few days ago

**Review by adlrocha — APPROVED** [review]
Nice. Thank you for the fix. Check the comments, see if they make sense and feel free to merge

File: src/lotus/message/ipc.rs (hunk 1 above)
Comment (cryptoAtwill) [reply src/lotus/message/ipc.rs:60]: TokenAmount does not implement `Serialize`, without the `serialize_with`, it's serialized into byte array which is definitely hard to read.

File: src/lotus/message/tests.rs (hunk 2 above)
Comment (cryptoAtwill) [reply src/lotus/message/tests.rs:109]: yep, tested, it should be ok

**Pull request merged**